packages/block/src/block.ts

```diff
@@ -62,14 +63,15 @@ export class Block {
   private cache: {
     txTrieRoot?: Uint8Array
   } = {}
+  private static keccakFunction: ((msg: Uint8Array) => Uint8Array) | undefined
 
   /**
    * Returns the withdrawals trie root for array of Withdrawal.
    * @param wts array of Withdrawal to compute the root of
    * @param optional emptyTrie to use to generate the root
    */
-  public static async genWithdrawalsTrieRoot(wts: Withdrawal[], emptyTrie?: Trie) {
-    const trie = emptyTrie ?? new Trie()
+  public static async genWithdrawalsTrieRoot(wts: Withdrawal[], emptyTrie: Trie) {
```

Comment:
This changes our API in a backwards-incompatible way and will break people's code, please revert.

Reply:
Seeing a bit where this is coming from, so the `new Trie()` would obviously not take a common here (and so not be able to leverage the WASM crypto if trie is not provided. But that's something we need to live with right now, then one just needs to pass the trie if one wants the crypto.

We can add this to #3216, have done so.

Otherwise have tackled here (push later).